### PR TITLE
xdg_shell support

### PR DESCRIPTION
**Pull request by yol**
@FernetMenta please take a look, especially the CMake stuff

### CODE REVIEW

**Comment by yol** [comment]
You surely mean `if(NOT CMAKE_VERSION VERSION_LESS 3.4)`?

**Comment by FernetMenta** [comment]
@pkerling my knowledge about cmake is limited. @hudokkow thanks for helping here. are the cmake changes ok?

**Comment by hudokkow** [comment]
Sorry @pkerling. Yes, you're correct. That's what I get for looking at code after a few beers. I'll have another look later today.

File: xbmc/windowing/wayland/WinSystemWayland.cpp [hunk 1]
```diff
@@ -289,46 +364,62 @@ bool CWinSystemWayland::SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, boo
...
+    g_graphicsContext.ResetOverscan(res);
+    CDisplaySettings::GetInstance().ApplyCalibrations();
+  }
+  
   // Force resolution update
+  // SetVideoResolution() automatically delegates to main thread via internal
+  // message if called from other threads
+  // This will call SetFullScreen() with the new resolution, which also updates
+  // the size of the egl_window etc.
+  // The call always blocks, so the configuration lock must be released beforehand.
```
Comment (FernetMenta) [inline xbmc/windowing/wayland/WinSystemWayland.cpp:422]: having windowing calling back into graphicsConext is bot a good idea (I know this is done in other implemenations too)
at least you should add a todo note here. future reference implementations should be clean in this regard.

**Review by FernetMenta — APPROVED** [review]
good work (I don't claim that I understand the details)

**Comment by hudokkow** [comment]
Sorry, I didn't had the time to look into cmake stuff more closely.
That said, I would suggest you drop the `if(CMAKE_VERSION VERSION_EQUAL 3.4.0 OR CMAKE_VERSION VERSION_GREATER 3.4.0)` check and bump `cmake_minimum_required(VERSION 3.1)` to 3.4 in https://github.com/xbmc/xbmc/blob/master/CMakeLists.txt#L1

There are already quite a few workarounds and semi-hacks in master because we're targeting such an old (2+ years) version. I'm using 3.8 on my machines. We might wanna ask @wsnipex's opinion. He's our packager and knows best about possible problems/complications/etc.

For reference, Ubuntu 16.04LTS ships with 3.5.1. Debian (jessie) ships with 3.0.2 but 3.6.2 is available in backports.

**Comment by yol** [comment]
@hudokkow Would you do that on master and remove the workarounds for <3.4? I don't want to mix it with the Wayland pull request if possible.

File: xbmc/windowing/wayland/WinSystemWayland.cpp (hunk 1 above)
Comment (yol) [reply xbmc/windowing/wayland/WinSystemWayland.cpp:422]: I also don't like it, but can't help it for now. I had to rework lots of this stuff for #8, TODO is included there.

**Comment by hudokkow** [comment]
Yap. Leave it as is. We can cleanup at a later stage.

**Pull request merged**